src/ru/skypro/StringListImpl.java

```diff
@@ -0,0 +1,168 @@
+package ru.skypro;
+
+
+import ru.skypro.execptions.NotFoundException;
+import java.util.Objects;
+
+
+public class StringListImpl implements StringList {
+
+    private String[] array = new String[10];
+    private int size = 0;
+
+    @Override
+    public String add(String item) {
+        if (size >= array.length) {
+            String[] extended = new String[array.length * 2];
+            System.arraycopy(array, 0, extended, 0,array.length);
+            array = extended;
+        }
+        array[size] = item;
+        size++;
+        return item;
+    }
+
+    @Override
+    public String add(int index, String item) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = item;
+        return item;
+    }
+
+    @Override
+    public String set(int index, String item) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+        String replaced = get(index);
+        array[index] = item;
+        return replaced;
+    }
+
+    @Override
+    public String remove(String item) {
+        int num = -1;
+        for (int i = 0; i < size; i++) {
+            if (item.equals(array[i])) {
+                num = i;
+                break;
+            }
+        }
+        if (num != -1) {
+            remove(num);
+        } else {
+            throw new NotFoundException("Was not found");
+        }
+        return item;
+    }
+
+    @Override
+    public String remove(int index) {
+        String element = get(index);
+        for (int i = index; i < size - 1; i++) {
+            array[i] = array[i + 1];
+            if (element.equals(array[i])) {
+                throw new NotFoundException("Was not found");
+            }
+        }
+            size--;
```

Comment:
```suggestion
            array[size - 1] = null;
            size--;
```
стоит обнулить последний элемент, поскольку он уже будет недоступен